### PR TITLE
Firstuserapi and enrollmentapi orm work

### PR DESCRIPTION
**Pull request by karppo**

### CODE REVIEW

File: src/rasenmaeher_api/db/enrollments.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ class EnrollmentPool(ORMBaseModel):  # pylint: disable=R0903
     owner = sa.Column(saUUID(), sa.ForeignKey(Person.pk), nullable=False)  # whos mainly responsible
     active = sa.Column(sa.Boolean(), nullable=False, default=True)
     extra = sa.Column(JSONB, nullable=False, server_default="{}")  # Passed on to the enrollments
+    invitecode = sa.Column(sa.String(), nullable=True)
```
Comment (rambo) [inline src/rasenmaeher_api/db/enrollments.py:31]: If we do lookups based on this it should be indexed and if we expect to only get one result per code it should also be unique

File: src/rasenmaeher_api/db/enrollments.py [hunk 2]
```diff
@@ -37,11 +38,45 @@ async def create_enrollment(self, callsign: str) -> "Enrollment":
...
+        return cast(EnrollmentPool, obj)
+
+    @classmethod
+    async def create_or_update_invitecode_for_callsign(
+        cls, callsign: str, extra: Optional[Dict[str, Any]] = None
+    ) -> Self:
+        """Create enrollment pool owned by callsign, or update invitecode for existing one"""
+        _owner = await Person.by_callsign(callsign=callsign)
+
+        _invitecode = "".join(secrets.choice(CODE_ALPHABET) for _ in range(CODE_CHAR_COUNT))
```
Comment (rambo) [inline src/rasenmaeher_api/db/enrollments.py:69]: only start variable with underscore if it's not meant to be used right now but will be Real Soon Now. If it's not meant to be used at all in general either assign it to just "_" (if it has to be assigned, like fixtures in unit tests) or don't assign it at all.

File: src/rasenmaeher_api/db/people.py [hunk 3]
```diff
@@ -22,9 +23,11 @@
 from ..prodcutapihelpers import post_to_all_products
 
 LOGGER = logging.getLogger(__name__)
+CODE_CHAR_COUNT = 12
+CODE_ALPHABET = string.ascii_uppercase + string.digits
```
Comment (rambo) [inline src/rasenmaeher_api/db/people.py:27]: Remove, these are not used for anything here.

Also should probably have just imported them from enrollments in the first place.

File: tests/test_firstuser.py [hunk 4]
```diff
@@ -4,100 +4,78 @@
...
-    /firstuser/enable
     Result should be successful
     """
-    app_client = tilauspalvelu_jwt_client
-    json_dict: Dict[Any, Any] = {"permit_str": settings.sqlite_init_management_hash}
-    resp = await app_client.post("/api/v1/firstuser/enable", json=json_dict)
+    json_dict: Dict[Any, Any] = {"work_id": "some_other_admin_ukkeli"}
+    resp = await tilauspalvelu_jwt_admin_client.post("/api/v1/firstuser/add-admin", json=json_dict)
+    resp_dict: Dict[Any, Any] = resp.json()
+    print(resp_dict)
```
Comment (rambo) [inline tests/test_firstuser.py:80]: Always use LOGGER instead of print

File: tests/test_firstuser.py [hunk 5]
```diff
@@ -4,100 +4,78 @@
...
-    /firstuser/disable
-    Result should be successful
+    Firstuser route fail using incorrect code
     """
-    app_client = tilauspalvelu_jwt_client
-    json_dict: Dict[Any, Any] = {"permit_str": settings.sqlite_init_management_hash}
-    resp = await app_client.post("/api/v1/firstuser/disable", json=json_dict)
+    resp = await unauth_client.get("/api/v1/firstuser/check-code?temp_admin_code=thiswontdo")
+    resp_dict: Dict[Any, Any] = resp.json()
+    print(resp_dict)
```
Comment (rambo) [inline tests/test_firstuser.py:66]: Always use LOGGER instead of print

File: tests/test_firstuser.py [hunk 6]
```diff
@@ -4,100 +4,78 @@
...
-    app_client = tilauspalvelu_jwt_client
-    json_dict: Dict[Any, Any] = {"temp_admin_code": settings.sqlite_first_time_user_hash, "work_id": "testikoira123"}
-    resp = await app_client.post("/api/v1/firstuser/delete-admin", json=json_dict)
-    assert resp.status_code == 200
+    # Create new admin user, the user already exits.. should not be 200
+    unauth_client.headers.update({"Authorization": f"Bearer {tmp_jwt}"})
+    json_dict = {"work_id": "nahkaesa"}
+    resp = await unauth_client.post("/api/v1/firstuser/add-admin", json=json_dict)
+    resp_dict = resp.json()
+    print(resp_dict)
```
Comment (rambo) [inline tests/test_firstuser.py:55]: Always use LOGGER instead of print

**Review by rambo — CHANGES_REQUESTED** [review]
initial changes, need to look into this locally in more detail

File: src/rasenmaeher_api/web/api/enrollment/schema.py [hunk 7]
```diff
@@ -206,19 +41,17 @@ class EnrollmentStatusIn(BaseModel, extra=Extra.forbid):  # pylint: disable=too-
 class EnrollmentStatusOut(BaseModel):  # pylint: disable=too-few-public-methods
     """Enrollment status check schema"""
 
-    status: str
+    status: int
     work_id: str
```
[the remark this answers is not in the document]
Comment (karppo) [reply src/rasenmaeher_api/web/api/enrollment/schema.py]: work_id's are now callsigns

File: tests/test_firstuser.py (hunk 4 above)
Comment (karppo) [reply tests/test_firstuser.py]: changed to LOGGER

File: src/rasenmaeher_api/db/enrollments.py (hunk 1 above)
Comment (karppo) [reply src/rasenmaeher_api/db/enrollments.py]: This came in from firstuserapi_orm_rambo

File: src/rasenmaeher_api/db/people.py (hunk 3 above)
Comment (karppo) [reply src/rasenmaeher_api/db/people.py]: Removed, or was removed by cherrypics from firstuserapi_orm_rambo

File: src/rasenmaeher_api/db/enrollments.py (hunk 2 above)
Comment (karppo) [reply src/rasenmaeher_api/db/enrollments.py:137]: Got it. Did a bit of renaming here and there.

**Review by rambo — APPROVED** [review]
Letsgooo!

**Pull request merged**